### PR TITLE
Create prow/deploy.sh, run at every commit

### PR DESCRIPTION
**Pull request by fejta**
This will allow us to automatically deploy prow by merging a PR

Tested locally with:
```console
$ docker run --rm=true --entrypoint prow/deploy.sh -v $PWD:/whatever -w /whatever -v /var/all-your-secrets-belong-to-us:/creds -e GOOGLE_APPLICATION_CREDENTIALS=/creds/secret.json
```

/assign @cjwagner

### CODE REVIEW

File: prow/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  local proj=$1
+  local zone=$2
+  local cluster=$3
+  local context="gke_${proj}_${zone}_${cluster}"
+  echo -n "$(color-context "$context") "
+  kubectl config get-contexts "$context" &> /dev/null && return 0
+  shift
+  echo "$(color-missing MISSING)"
+  gcloud container clusters get-credentials --project="$proj" --zone="$zone" "$cluster"
+  kubectl config get-contexts "$context" > /dev/null
```
Comment (cjwagner) [inline prow/deploy.sh]: I don't understand what is happening following line 50. Won't this always print "MISSING"? Why is `shift` needed?

File: testgrid/config.yaml [hunk 2]
```diff
@@ -6654,6 +6656,11 @@ dashboards:
     test_group_name: post-test-infra-push-prow
     code_search_url_template:
       url: https://github.com/kubernetes/test-infra/compare/<start-custom-0>...<end-custom-0>
+  - name: deploy-prow
+    description: deploys the configured version of prow by running deploy.sh
+    test_group_name: post-test-infra-deploy-prow
+    code_search_url_template:
+      url: https://github.com/kubernetes/test-infra/compare/<start-custom-0>...<end-custom-0>
```
Comment (cjwagner) [inline testgrid/config.yaml]: Should we add an alert?

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 3]
```diff
@@ -23,6 +23,25 @@ postsubmits:
         secret:
           secretName: oauth-token
   kubernetes/test-infra:
+  - name: post-test-infra-deploy-prow
```
Comment (cjwagner) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: This should only run on the `master` branch.

Comment (fejta) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: Done

File: prow/deploy.sh (hunk 1 above)
Comment (fejta) [reply prow/deploy.sh]: Done, removed unnecessary `shift`

If the context already exists we exit the function on L54 `kubectl get && return 0`

File: testgrid/config.yaml (hunk 2 above)
Comment (fejta) [reply testgrid/config.yaml]: I don't think so, too slow.

Ideally we provide a way to report back to the PR with the job it kicked off

File: prow/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  local proj=$1
+  local zone=$2
+  local cluster=$3
+  local context="gke_${proj}_${zone}_${cluster}"
+  echo -n " $(color-context "$context")"
+  kubectl config get-contexts "$context" &> /dev/null && return 0
+  echo ": $(color-missing MISSING), getting credentials..."
+  gcloud container clusters get-credentials --project="$proj" --zone="$zone" "$cluster"
+  kubectl config get-contexts "$context" > /dev/null
+  echo -n "Ensuring contexts exist:"
```
Comment (cjwagner) [inline prow/deploy.sh]: nit: I think this is redundant.

**Comment by fejta** [comment]
or rather, EXCITED TO TRY THIS OUT with https://github.com/kubernetes/test-infra/pull/10101

**Comment by fejta** [comment]
Worked the first time, thank you `docker run` test :)

**Pull request merged**